pkg/networkservice/common/localbypass/server.go

```diff
@@ -21,61 +21,43 @@ package localbypass
 import (
 	"context"
 	"net/url"
-	"sync"
 
-	"github.com/networkservicemesh/sdk/pkg/registry/common/localbypass"
+	"github.com/networkservicemesh/sdk/pkg/registry/memory"
 
 	"github.com/golang/protobuf/ptypes/empty"
-
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
-	"github.com/networkservicemesh/api/pkg/api/registry"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
 type localBypassServer struct {
-	// Map of names -> *url.URLs for local bypass to file sockets
-	sockets sync.Map
+	storage *memory.Storage
 }
 
 // NewServer - creates a NetworkServiceServer that tracks locally registered Endpoints substitutes their
 //             passed endpoint_address with clienturl.ClientURL(ctx) used to connect to them.
-//             - server - *registry.NetworkServiceRegistryServer.  Since registry.NetworkServiceRegistryServer is an interface
-//                        (and thus a pointer) *registry.NetworkServiceRegistryServer is a double pointer.  Meaning it
-//                        points to a place that points to a place that implements registry.NetworkServiceRegistryServer
-//                        This is done so that we can return a registry.NetworkServiceRegistryServer chain element
-//                        while maintaining the NewServer pattern for use like anything else in a chain.
-//                        The value in *server must be included in the registry.NetworkServiceRegistryServer listening
-//                        so it can capture the registrations.
-func NewServer(registryServer *registry.NetworkServiceRegistryServer) networkservice.NetworkServiceServer {
-	rv := &localBypassServer{}
-	*registryServer = localbypass.NewNetworkServiceRegistryServer(rv)
+func NewServer(storage *memory.Storage) networkservice.NetworkServiceServer {
```

Comment:
Could we please *not* multiple the number of APIs wandering around?  I'm actually not at all happy about memory.Storage currently... I'd prefer not to propogate it to additional packages.

The original pattern involved *returning* the registry piece to the caller via the caller providing a place to provide it.  That kept us to our stock APIs.  Whatever it is you need to do with that registry chain element, it was registry chain element.  Now a developer has to learn about a memory.Storage.

Reply:
What if we simply had localbypass.NewServer keep its signature but write a pointer to its internally created memory registry server.  That way you don't have to test the registry bit (its tested already).

Reply:
Both variants not looking good, let's wait for registry simplification to eliminate localbypass at all.

Reply:
>Could we please not multiple the number of APIs wandering around? I'm actually not at all happy about memory.Storage currently... I'd prefer not to propogate it to additional packages.

I think we'll get rid of `memory.Storage` when registry API simplification will be completed.